src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderValidationConsensusRuleBase.cs

```diff
@@ -0,0 +1,58 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Networks;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.ProvenHeaderRules
+{
+    /// <summary>
+    /// Base rule to be used by all proven header validation rules. 
+    /// </summary>
+    /// <seealso cref="Stratis.Bitcoin.Consensus.Rules.HeaderValidationConsensusRule" />
+    public abstract class ProvenHeaderValidationConsensusRuleBase : HeaderValidationConsensusRule
+    {
+        /// <summary>Allow access to the POS parent.</summary>
+        protected PosConsensusRuleEngine PosParent;
+
+        /// <inheritdoc />
+        public override void Initialize()
+        {
+            this.PosParent = this.Parent as PosConsensusRuleEngine;
+
+            Guard.NotNull(this.PosParent, nameof(this.PosParent));
+        }
+
+        /// <summary>
+        /// Determines whether proven headers are activated based on the proven header activation height and applicable network.
+        /// </summary>
+        /// <param name="height">The height of the header.</param>
+        /// <returns>
+        /// <c>true</c> if proven header height is past the activation height for the corresponding network;
+        /// otherwise, <c>false</c>.
+        /// </returns>
+        public bool IsProvenHeaderActivated(int height)
+        {
+            if (this.PosParent.Network is StratisMain)
```

Comment:
Right... this becomes a StratisNetwork rule only, in this case we need to change the rule to be `StratisProvenHeaderValidationConsensusRuleBase ` and only inject it for stratis network.

This is probably wrong.
Any other network then stratis PH will be activated by default.

So the solution is to use overrides, but this is an abstract class so the logic `IsProvenHeaderActivated` should not be here at all but in some rule that is CheckProvenHeaderActivation (or something).

Reply:
I am confused with this. What exactly should we do?